### PR TITLE
String container stats from the VSD branch

### PR DESCRIPTION
**Pull request by martin-cs**
Like https://github.com/diffblue/cbmc/pull/5202 and https://github.com/diffblue/cbmc/pull/5203 this is taken from the variable-sensitivity-domain branch and split out as separate commits.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/util/memory_units.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  static const char *si_kibitye_symbol;
+  static const char *si_mebibyte_symbol;
+  static const char *si_gibibyte_symbol;
+
+private:
+  std::size_t bytes;
+  memory_sizet(std::size_t bytes);
+};
+
+#endif // CPROVER_UTIL_MEMORY_UNITS_H
```
Comment (peterschrammel) [inline src/util/memory_units.h]: missing newline

File: src/util/memory_units.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#ifndef CPROVER_UTIL_MEMORY_UNITS_H
+#define CPROVER_UTIL_MEMORY_UNITS_H
+
+#include <cstddef>
+#include <string>
+
+struct memory_sizet
```
Comment (peterschrammel) [inline src/util/memory_units.h]: class ?

File: src/util/memory_units.cpp [hunk 3]
```diff
@@ -80,7 +80,7 @@ std::string memory_sizet::to_string() const
     {
       out << ' ';
     }
-    out << kib << si_kibitye_symbol;
+    out << kib << si_kibibyte_symbol;
```
Comment (peterschrammel) [inline src/util/memory_units.cpp]: memory_unit changes should be in previous commit

File: src/util/Makefile [hunk 4]
```diff
@@ -45,6 +45,7 @@ SRC = allocate_objects.cpp \
       mathematical_expr.cpp \
       mathematical_types.cpp \
       memory_info.cpp \
+      memory_units.cpp \
```
Comment (peterschrammel) [inline src/util/Makefile]: squash with initial memory_units commit

File: src/util/memory_units.cpp [hunk 5]
```diff
@@ -1,3 +1,11 @@
+/*******************************************************************\
+
+Module: Memory units
```
Comment (peterschrammel) [inline src/util/memory_units.cpp]: Squash

File: src/util/memory_units.h (hunk 2 above)
Comment (martin-cs) [reply src/util/memory_units.h]: Sure, why not?

File: src/util/memory_units.cpp (hunk 3 above)
Comment (martin-cs) [reply src/util/memory_units.cpp]: Refactored and sqashed.

File: src/util/Makefile (hunk 4 above)
Comment (martin-cs) [reply src/util/Makefile]: Squashed.

File: src/util/memory_units.cpp (hunk 5 above)
Comment (martin-cs) [reply src/util/memory_units.cpp]: Squashed

**Comment by martin-cs** [comment]
All review comments resolved.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 3209bd86513dcc1375f57e3c2bcc0385de5e0a5c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/140864409

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]
I’m OK with this _per se_, but it doesn’t seem to be used anywhere at the moment?

**Comment by martin-cs** [comment]
@hannes-steffenhagen-diffblue It is used in the fifth VSD patch which is still to be PR'd.  I wanted to get the foundations reviewed and ready first.

**Pull request closed without merging**